test/app-tap/gh-2717-no-quit-sigint.test.lua

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env tarantool
+
+local tap = require('tap')
+local popen = require('popen')
+local process_timeout = require('process_timeout')
+local fio = require('fio')
+local clock = require('clock')
+
+--
+-- gh-2717: tarantool console quit on sigint
+--
+local file_read_timeout = 60.0
+local file_read_interval = 0.2
+local file_open_timeout = 60.0
+local prompt = 'tarantool> '
+
+local TARANTOOL_PATH = arg[-1]
+local test = tap.test('gh-2717')
+
+test:plan(3)
+
+local ph = popen.shell('ERRINJ_STDIN_ISATTY=1 ' .. TARANTOOL_PATH .. ' -i 2>&1', 'r')
```

Comment:
Here I got corrupted terminal state after the test (even if I run it under test-run). Please, consider hiding real stdin from the child process:

```diff
diff --git a/test/app-tap/gh-2717-no-quit-sigint.test.lua b/test/app-tap/gh-2717-no-quit-sigint.test.lua
index ca28f8bf4..2cd7c8f45 100755
--- a/test/app-tap/gh-2717-no-quit-sigint.test.lua
+++ b/test/app-tap/gh-2717-no-quit-sigint.test.lua
@@ -19,7 +19,18 @@ local test = tap.test('gh-2717')
 
 test:plan(3)
 
-local ph = popen.shell('ERRINJ_STDIN_ISATTY=1 ' .. TARANTOOL_PATH .. ' -i 2>&1', 'r')
+-- We set stdin to /dev/null to don't let the child process
+-- manage terminal options. In may leave the terminal in an
+-- inconsistent state and requires to call `reset` after the
+-- test.
+local ph = popen.new({'ERRINJ_STDIN_ISATTY=1 ' .. TARANTOOL_PATH .. ' -i 2>&1'}, {
+    shell = true,
+    setsid = true,
+    group_signal = true,
+    stdin = popen.opts.DEVNULL,
+    stdout = popen.opts.PIPE,
+    stderr = popen.opts.DEVNULL,
+})
 assert(ph, 'process is not up')
 
 local start_time = clock.monotonic()
```

Reply:
Thanks for the patch! Fixed.